### ISSUE TITLE
support for exec

### ISSUE THREAD

**Issue by jonasrauber** [aio-libs/aiodocker#174]
It would be great if aiodocker would support `exec` similar to the `docker exec` command and the `exec_run` method in [docker-py](https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.exec_run).

**Comment by barrachri** [comment]
@jonasrauber that would be great!

Happy to receive a PR in case 👍

**Comment by achimnol** [comment]
Just running a command as a background process would work.

There is a issue, though: TCP-over-HTTP(S) emulation in aiohttp.  
Docker uses a trick to "upgrade" an HTTP connection to TCP (not WebSocket!) to attach the pseudo-tty of containers in `docker attach` and `docker exec -i -t` commands.
Currently, aiohttp does not support such cases.

**Comment by hirokiky** [comment]
I created `container.exec` equivalent, but it's implementation can't take standard input
(as @achimnol said).

like:

```python
    data = await container.docker._query_json(
        f"containers/{container._id}/exec",
        method='POST',
        data={ ... },
    )
    exec_id = data['Id']
    # docker._query_json will call parse_response
    # and it will cause error for vnd.docker.raw-stream content-type
    response = await container.docker._query(
        f"exec/{exec_id}/start",
        method='POST',
        headers={"content-type": "application/json"},
        data=json.dumps({ ... }),
    )
    result = await response.read()
    await response.release()
```

**Comment by hirokiky** [comment]
Working on this branch https://github.com/hirokiky/aiodocker/tree/feature-exec-run

**Comment by SoerenBusse** [comment]
Any new information when this change will be merged?

**Comment by asvetlov** [comment]
There are 2 PRs.
@SoerenBusse which one are you talking about?

**Comment by SoerenBusse** [comment]
Hey there, I cannot pass judgement on which of them is better. However it would be awesome if one of them would be merged soon, because I need the exec method in a project :)

**Comment by asvetlov** [comment]
Fixed by #415
